bugbug/similarity.py

```diff
@@ -626,6 +628,65 @@ def get_distance(self, query1, query2):
         raise NotImplementedError
 
 
+class ElasticSearchSimilarity(BaseSimilarity):
+    def __init__(self, cleanup_urls=True, nltk_tokenizer=False):
+        super().__init__(cleanup_urls=cleanup_urls, nltk_tokenizer=nltk_tokenizer)
+        self.elastic_search = Elasticsearch()
+        assert (
+            self.elastic_search.ping()
+        ), "Check if Elastic Search Server is running by visiting http://localhost:9200"
+
+    def get_text(self, bug):
+        return "{} {}".format(
+            bug["summary"], " ".join([comment["text"] for comment in bug["comments"]])
+        )
+
+    def text_preprocess(self, text):
+        for func in self.cleanup_functions:
+            text = func(text)
+
+        text = re.sub("[^a-zA-Z0-9]", " ", text)
+        return text
+
+    def make_documents(self):
+        for bug in bugzilla.get_bugs():
+            yield {
+                "_index": "bugbug",
+                "_type": "_doc",
+                "bug_id": bug["id"],
+                "description": self.text_preprocess(self.get_text(bug)),
+            }
+
+    def index(self):
+        self.elastic_search.indices.delete(index="bugbug", ignore=[400, 404])
+        bulk(self.elastic_search, self.make_documents())
+
+    def get_similar_bugs(self, query):
+        find_similar = self.text_preprocess(self.get_text(query))
```

Comment:
Have you tried considering summary and comments separately?

Reply:
nope, i will give it a try today

Reply:
The results are almost same.